### PR TITLE
Start module names in atmosphere diagnostics with "mpas_"

### PR DESCRIPTION
**Pull request by gdicker1**
This change prevents name collisions from occurring when MPAS-A is compiled as part of a larger system (e.g. coupled as a dynamical core in an earth system model). The diagnostics is one portion of the MPAS-A code where the modules don't already start with "mpas_", and where similarly named modules could more easily occur than others.

NOTE: none of this is of any concern in stand-alone MPAS-A since name collisions would prevent builds from succeeding.

### CODE REVIEW

**Comment by gdicker1** [comment]
This has been tested on Cheyenne by building and running a [FHS94](https://www.cesm.ucar.edu/models/simpler-models/changedycore.html) compset of CESM using a 480km MPAS grid. Adding these changes to the MPAS code did not cause any issues with the build/run.

**Comment by mgduda** [comment]
@gdicker1 In the description of the PR, I think we had discussed the possibility that the statement "To ensure the functions in atmosphere diagnostics have unique names when compiled..." could be confusing: even before this PR, within stand-alone MPAS-A, the name "cloud_diagnostics" is unique, for example.

**Comment by gdicker1** [comment]
@mgduda let me know what you think!

**Comment by mgduda** [comment]
> @mgduda I edited the PR description, let me know what you think!

The new PR description looks good -- thanks! I had just one other comment on the README.

**Review by mgduda — CHANGES_REQUESTED** [review]

File: src/core_atmosphere/diagnostics/README [hunk 1]
```diff
@@ -9,7 +9,8 @@ generally required to implement a diagnostic.
    Registry_diagnostics.xml.
 
 2) Create a new module for the diagnostic; the "mpas_atm_diagnostic_template.F" 
-   module file may be used as a template.
+   module file may be used as a template. The module name is expected to begin
+   with "mpas_" to prevent name conflicts.
```
[the remark this answers is not in the document]
Comment (gdicker1) [reply src/core_atmosphere/diagnostics/README]: We did. Does the force push to bcfa498 solve this for you?

(I force pushed to split a commit that both updated the README and a module name)

[the remark this answers is not in the document]
Comment (mgduda) [reply src/core_atmosphere/diagnostics/README]: The `README` looks good now. Thanks!

**Review by mgduda — APPROVED** [review]

**Pull request merged**